modules/gapi/include/opencv2/gapi/gcommon.hpp

```diff
@@ -94,6 +94,25 @@ enum class GShape: int
     GFRAME,
 };
 
+namespace gapi {
+namespace s11n {
+struct IOStream;
+struct IIStream;
+
+namespace detail {
+template<typename T, typename U> struct wrap_serialize {
+    static std::function<void(gapi::s11n::IOStream&, const util::any&)> serialize;
+};
+
+template<typename T, typename U>
+std::function<void(gapi::s11n::IOStream&, const util::any&)>
+wrap_serialize<T, U>::serialize = nullptr;
```

Comment:
It shouldn't be like this.

Reply:
I'd expect you to have a look on `call_helper` thing from our kernels API to get the idea.

Reply:
Reworked.